### PR TITLE
Exit early on error flag. Enable coverage reports on failure case

### PR DESCRIPTION
**Pull request by ZacharyBenamram**
## PR Summary

*Replace this text with a description of what this PR is accomplishing*

 - **Type:** Features|Bugfix|Improvements|Internal
 - **Issue Link:** https://github.com/blend/go-sdk/issues/??
 - **Intended Change Level:** patch|minor|major

#### Reviewers:

Reviewers keep the following in mind while reviewing this PR, and provide an assessment of them in your approval comment:

- Does the "Intended Change Level" above match the actual behavior of this PR?
- Is this PR following patterns set forth in the package (if modifying a package), or the go-sdk design patterns if implementing a new package from scratch?
- Does this require a backport to LTS versions? If so ping, let the contributors group know.

### CODE REVIEW

File: cmd/coverage/main.go [hunk 1]
```diff
@@ -134,11 +134,10 @@ func walkPath(walkedPath string, fullCoverageData *os.File) []error {
 	maybeFatal(filepath.Walk(rootPath, func(currentPath string, info os.FileInfo, fileErr error) error {
 		packageCoverReport, err := getPackageCoverage(currentPath, info, fileErr)
 
-		if err != nil && *exitOnFirstCoverageFailure {
-			return err
-		}
-
-		if err != nil && len(packageCoverReport) > 0 {
+		if err != nil && (*exitOnFirstCoverageFailure || len(packageCoverReport) == 0) {
+			if *exitOnFirstCoverageFailure || len(packageCoverReport) == 0 {
```
Comment (dhermes) [inline cmd/coverage/main.go:138]: These two lines are "too complex". Can you re-factor / turn it into a helper function?

**Comment by ZacharyBenamram** [comment]
failure case: http://jenkins.sandbox.blendlabs.com/job/deployinator-ghprb/3396/console

**Comment by ZacharyBenamram** [comment]
success case: http://jenkins.sandbox.blendlabs.com/job/deployinator-ghprb/3397/console

**Review by dhermes — APPROVED** [review]
LGTM

**Pull request merged**